commands/orchestrationcommands/stackeventcommands/listresource.go

```diff
@@ -0,0 +1,117 @@
+	"github.com/rackspace/rack/handler"
+	"github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+	"github.com/rackspace/rack/internal/github.com/fatih/structs"
+	osStackEvents "github.com/rackspace/rack/internal/github.com/rackspace/gophercloud/openstack/orchestration/v1/stackevents"
+	"github.com/rackspace/rack/internal/github.com/rackspace/gophercloud/rackspace/orchestration/v1/stackevents"
+	"github.com/rackspace/rack/util"
+)
+
+var listResource = cli.Command{
+	Name:        "list-resource",
+	Usage:       util.Usage(commandPrefix, "list-resource", "[--name <stackName> | --id <stackID>] --resource <resourceName>"),
+	Description: "Lists events for a specified stack resource",
+	Action:      actionListResource,
+	Flags:       commandoptions.CommandFlags(flagsListResource, keysListResource),
+	BashComplete: func(c *cli.Context) {
+		commandoptions.CompleteFlags(commandoptions.CommandFlags(flagsListResource, keysListResource))
+	},
+}
+
+func flagsListResource() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "[optional; required if `id` isn't specified] The stack name.",
+		},
+		cli.StringFlag{
+			Name:  "id",
+			Usage: "[optional; required if `name` isn't specified] The stack id.",
+		},
+		cli.StringFlag{
+			Name:  "resource",
+			Usage: "[required] The resource name.",
+		},
+	}
+}
+
+type paramsListResource struct {
+	stackName    string
+	stackID      string
+	resourceName string
+}
+
+var keysListResource = []string{"ResourceName", "Time", "ResourceStatusReason", "ResourceStatus", "PhysicalResourceID", "ID"}
+
+type commandListResource handler.Command
+
+func actionListResource(c *cli.Context) {
+	command := &commandListResource{
+		Ctx: &handler.Context{
+			CLIContext: c,
+		},
+	}
+	handler.Handle(command)
+}
+
+func (command *commandListResource) Context() *handler.Context {
+	return command.Ctx
+}
+
+func (command *commandListResource) Keys() []string {
+	return keysListResource
+}
+
+func (command *commandListResource) ServiceClientType() string {
+	return serviceClientType
+}
+
+func (command *commandListResource) HandleFlags(resource *handler.Resource) error {
+	if err := command.Ctx.CheckFlagsSet([]string{"resource"}); err != nil {
+		return err
+	}
+
+	c := command.Ctx.CLIContext
+	name := c.String("name")
+	id := c.String("id")
+	name, id, err := stackcommands.IDAndName(command.Ctx.ServiceClient, name, id)
+	if err != nil {
+		return err
+	}
+	resource.Params = &paramsListResource{
+		stackName:    name,
+		stackID:      id,
+		resourceName: c.String("resource"),
+	}
+	return nil
+}
+
+func (command *commandListResource) Execute(resource *handler.Resource) {
+	params := resource.Params.(*paramsListResource)
+	stackName := params.stackName
+	stackID := params.stackID
+	resourceName := params.resourceName
+
+	pager := stackevents.ListResourceEvents(command.Ctx.ServiceClient, stackName, stackID, resourceName, nil)
+	pages, err := pager.AllPages()
```

Comment:
Have a look at other `list` commands to see how we handle paging with the `all-pages` flag.